funnel/views/proposal.py

```diff
@@ -294,19 +286,17 @@ def proposal_view(profile, space, proposal):
                 flash(_("No such comment"), 'error')
             return redirect(proposal.url_for(), code=303)
     links = [Markup(linkify(unicode(escape(l)))) for l in proposal.links.replace('\r\n', '\n').split('\n') if l]
-    if not proposal.state.DRAFT:
-        statusform = ProposalStatusForm(status=proposal.state.value)
-    else:
-        statusform = None
+
+    transitionform = ProposalTransitionForm(obj=proposal)
 
     proposal_move_form = None
-    if 'move-proposal' in space.permissions(g.user, g.permissions):
+    if proposal.current_roles.admin:
```

Comment:
This thing is bothering me a bit. Neither the old permission model nor the role model adequately describes what is needed. Temporary fix (still incomplete):

1. Add a `move_to` method to the object. It takes a new parent and moves to it, but does not validate if the user has permission to do this.
2. Decorate this with `@with_roles(call={'admin'})`
3. On this line here, check `if 'move_to' in proposal.current_access():`
4. After form validation, call `proposal.current_access().move_to(form.new_parent.data)`
5. TODO: Find a way to declaratively associate `ProposalMoveForm` with this method, mapping form fields to method parameters.

Reply:
Should I do this in this PR? or a separate one?

Reply:
We can't merge this PR until we review transitions with editorial, so you might as well incorporate these.